crates/sui-storage/src/indexes.rs

```diff
@@ -19,37 +20,59 @@ use typed_store::rocks::DBMap;
 use typed_store::traits::DBMapTableUtil;
 use typed_store::traits::Map;
 
+use crate::default_db_options;
+
 #[derive(DBMapUtils)]
 pub struct IndexStore {
     /// Index from sui address to transactions initiated by that address.
-    #[options(cache_capacity = 1_000_000)]
+    #[default_options_override_fn = "transactions_from_addr_table_default_config"]
     transactions_from_addr: DBMap<(SuiAddress, TxSequenceNumber), TransactionDigest>,
 
     /// Index from sui address to transactions that were sent to that address.
-    #[options(cache_capacity = 1_000_000)]
+    #[default_options_override_fn = "transactions_to_addr_table_default_config"]
     transactions_to_addr: DBMap<(SuiAddress, TxSequenceNumber), TransactionDigest>,
 
     /// Index from object id to transactions that used that object id as input.
-    #[options(cache_capacity = 1_000_000)]
+    #[default_options_override_fn = "transactions_by_input_object_id_table_default_config"]
     transactions_by_input_object_id: DBMap<(ObjectID, TxSequenceNumber), TransactionDigest>,
 
     /// Index from object id to transactions that modified/created that object id.
-    #[options(cache_capacity = 1_000_000)]
+    #[default_options_override_fn = "transactions_by_mutated_object_id_table_default_config"]
     transactions_by_mutated_object_id: DBMap<(ObjectID, TxSequenceNumber), TransactionDigest>,
 
     /// Index from package id, module and function identifier to transactions that used that moce function call as input.
-    #[options(cache_capacity = 1_000_000)]
+    #[default_options_override_fn = "transactions_by_move_function_table_default_config"]
     transactions_by_move_function:
         DBMap<(ObjectID, String, String, TxSequenceNumber), TransactionDigest>,
 
     /// This is a map between the transaction digest and its timestamp (UTC timestamp in
     /// **milliseconds** since epoch 1/1/1970). A transaction digest is subjectively time stamped
     /// on a node according to the local machine time, so it varies across nodes.
     /// The timestamping happens when the node sees a txn certificate for the first time.
-    #[options(optimization = "point_lookup", cache_capacity = 1_000_000)]
+    #[default_options_override_fn = "timestamps_table_default_config"]
     timestamps: DBMap<TransactionDigest, u64>,
 }
 
+// These functions are used to initialize the DB tables
+fn transactions_from_addr_table_default_config() -> Options {
+    default_db_options(None, Some(1_000_000)).0
```

Comment:
Can we collapse the diff options into a fn such as `index_cache_option`?

Reply:
We actually can. But see my other comment on keeping it explicit to avoid confusing people. Now if the crate owner wants to condense them, they're welcome to.

Reply:
But yes your thinking is correct